golang/types/plugin.go

```diff
@@ -0,0 +1,45 @@
+package types
+
+// WaitForExpressionOpts defines options for waiting for a JavaScript expression to evaluate to true.
+type WaitForExpressionOpts struct {
+	Timeout  *float64 // Maximum time to wait in milliseconds
+	Interval *float64 // Interval between checks in milliseconds
+}
+
+// ScrollPosition is an alias for Point, representing a scroll position in a 2D space.
+type ScrollPosition = Point
+
+// PluginInterface defines the interface for a plugin that interacts with a web page.
+type PluginInterface interface {
+
+	// EvaluateExpression evaluates a given JavaScript expression with optional arguments.
+	EvaluateExpression(expression string, args ...any) (any, error)
```

Comment:
Don't make these public interface functions; we need to make the plugins compatible across multiple platforms (web/ mobile). Don't include platform specific functions or evaluation handlers. We'll do that internally within each plugin for the platform.